src/io/flutter/utils/MostlySilentOsProcessHandler.java

```diff
@@ -30,14 +32,33 @@
  * for more information.
  */
 public class MostlySilentOsProcessHandler extends OSProcessHandler {
+  private boolean softKill;
+
   public MostlySilentOsProcessHandler(@NotNull GeneralCommandLine commandLine)
+    throws ExecutionException {
+    this(commandLine, false);
+  }
+
+  public MostlySilentOsProcessHandler(@NotNull GeneralCommandLine commandLine, boolean softKill)
     throws ExecutionException {
     super(commandLine);
+    this.softKill = softKill;
   }
 
   @NotNull
   @Override
   protected BaseOutputReader.Options readerOptions() {
     return BaseOutputReader.Options.forMostlySilentProcess();
   }
+
+  @Override
+  protected void doDestroyProcess() {
+    final Process process = getProcess();
+    if (softKill && SystemInfo.isUnix && shouldDestroyProcessRecursively() && processCanBeKilledByOS(process)) {
+      UnixProcessManager.sendSigIntToProcessTree(process);
```

Comment:
Can we verify whether we were able to kill the process and fail if we weren't able to?

Reply:
`UnixProcessManager.sendSigIntToProcessTree` will log a warning message and return false if the process isn't killed - what's a reasonable action to take if that's the case? Throw an exception or call `super.doDestroyProcess`?

Reply:
Lets kill the process completely but also log a message to analytics so we can diagnose what went wrong.